common/core/tool_detection.cpp

```diff
@@ -0,0 +1,74 @@
+#include "common/core/tool_detection.hpp"
+
+#include <array>
+#include <span>
+
+using namespace tool_detection;
+
+constexpr static auto pipette_384_chan_z_bounds =
```

Comment:
I thought we were only going to be detecting differences between gripper vs pipette vs no instrument attached

Reply:
We have the means to figure out the type of pipette -[this sheet shows the different expected voltages ](https://opentrons.slack.com/files/U01SBA3K485/F02PHGXC9GU/slot_id_resistor_calculations)

Reply:
@Laura-Danielle we'll probably go that direction eventually but this part was already implemented so I stuck with it